core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -129,7 +130,7 @@ private[spark] class PipedRDD[T: ClassTag](
       override def run(): Unit = {
         val err = proc.getErrorStream
         try {
-          for (line <- Source.fromInputStream(err).getLines) {
+          for (line <- Source.fromInputStream(err)(StandardCharsets.UTF_8).getLines) {
```

Comment:
I think we've been using StandardCharsets pretty consistently for the constant (?). Or should.
It's a fair point about how to read the input from a local system process, though I still kinda imagine that UTF-8 is the safer default. It's nice to not make the behavior dependent on the local env. I think a well-behaved program will consistently output one encoding and that encoding is UTF-8. That is, just because the local env is Shift_JIS doesn't really suggest that apps on the machine are using Shift_JIS by default.